dbms/src/Flash/Coprocessor/DAGUtils.cpp

```diff
@@ -1437,4 +1437,26 @@ tipb::ScalarFuncSig reverseGetFuncSigByFuncName(const String & name)
     return func_name_sig_map[name];
 }
 
+size_t getMaxStreams(const Context & context)
+{
+    size_t max_streams = context.getSettingsRef().max_threads;
+    auto * dag_context = context.getDAGContext();
+    if (dag_context != nullptr)
+    {
+        if (!dag_context->isBatchCop() && !dag_context->isMPPTask())
+            /// for normal cop request, the max_streams should be 1
+            max_streams = 1;
+        else
+        {
+            if (dag_context->isTest())
+                max_streams = dag_context->initialize_concurrency;
+        }
```

Comment:
I think 
```
if (dag_context->isTest())
    max_streams = dag_context->initialize_concurrency;
else if (!dag_context->isBatchCop() && !dag_context->isMPPTask())
    max_streams = 1;
else
    max_streams = context.getSettingsRef().max_threads;
```
is better

Reply:
because there may be
`dag_context->isTest() && !dag_context->isBatchCop() && !dag_context->isMPPTask()`